R/generate_data.R

```diff
@@ -80,6 +80,38 @@ GenGeno <- function(
 }
 
 
+#' Filter Noncausal Variants
+#' 
+#' Remove a random fraction of variants, which are designated non-causal. 
+#' 
+#' @param anno (snps x 1) annotation vector.
+#' @param geno (n x snps) genotype matrix.
+#' @param prop_causal Proportion of variants which are causal.
+#' @return List containing the (n x snps) genotype matrix "geno" and the
+#' (snps x 1) annotation vector "anno".
+FilterGenos <- function(
+  anno,
+  geno,
+  prop_causal = 1.0
+) {
+  
+  p_noncausal <- 1.0 - prop_causal
+  which_drop <- stats::rbinom(length(anno), size = 1, prob = p_noncausal)
```

Comment:
Could you document why we use `binom` here to perform the selection?

Reply:
This was the first approach that came to mind when writing the function, but your completely right, binom is not needed here. Updated to simplify.